src/System.IO.FileSystem/tests/FileStream/ReadWriteSpan.netcoreapp.cs

```diff
@@ -0,0 +1,192 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public abstract class FileStream_ReadWrite_Span : FileSystemTest
+    {
+        protected abstract FileStream CreateFileStream(string path, FileMode mode, FileAccess access = FileAccess.ReadWrite);
+
+        [Fact]
+        public void DisposedStream_ReadWrite_Throws()
+        {
+            var fs = CreateFileStream(GetTestFilePath(), FileMode.Create);
+            fs.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => fs.Read(new Span<byte>(new byte[1])));
+            Assert.Throws<ObjectDisposedException>(() => fs.Write(new Span<byte>(new byte[1])));
```

Comment:
It's a good question, and I've been going back and forth on what to do about it.  Right now the base Read/Write span implementation special cases 0-byte reads/writes and treats them as nops, but thinking more about it, I think we should make them delegate to the array-based methods even in the 0-byte case.  Doing so gives the derived stream a chance to vet the operation, and in a case like this with exceptions, that actually matters.  With the current behavior, a 0-byte span read/write will throw an ObjectDisposedException if the stream is configured in sync mode (because then the stream's override is used), but doesn't throw if the stream is in async mode (because then its override delegates to the base implementation which is a nop for a 0-byte operation).  I'll submit a PR to make the change in coreclr.  If nothing else, since 0-byte reads/writes are relatively uncommon, we'll avoid a branch for an uncommon case.